Intersect.Server/Entities/BankInterface.cs

```diff
@@ -319,79 +322,106 @@ public bool TryDepositItem(int slot, int amount, int destSlot = -1, bool sendUpd
+                {
+                    // Try to find an open slot with the same item type and not at maximum stack size
+                    for (int i = 0; i < mMaxSlots; i++)
+                    {
+                        var bankItem = mBank[i];
+                        if (bankItem.ItemId == Guid.Empty ||
+                            (bankItem.ItemId != Guid.Empty && bankItem.ItemId == inventoryItemItemId &&
+                             amount + bankItem.Quantity < bankItem.Descriptor.MaxBankStack))
                         {
-                            if (itemBase.IsStackable)
-                            {
-                                PutItem(new Item(itemBase.Id, amount), destSlot, sendUpdate);
-                                mPlayer.TryTakeItem(itemBase.Id, amount, ItemHandling.Normal, sendUpdate);
-
-                                if (mGuild != null)
-                                {
-                                    DbInterface.Pool.QueueWorkItem(mGuild.Save);
-                                }
-
-                                return true;
-                            }
-                            else
-                            {
-                                PutItem(mPlayer.Items[slot], destSlot, sendUpdate);
-
-                                mPlayer.Items[slot].Set(Item.None);
-                                mPlayer.EquipmentProcessItemLoss(slot);
-
-                                if (sendUpdate)
-                                {
-                                    PacketSender.SendInventoryItemUpdate(mPlayer, slot);
-                                }
-
-                                if (mGuild != null)
-                                {
-                                    DbInterface.Pool.QueueWorkItem(mGuild.Save);
-                                }
-
-                                return true;
-                            }
+                            destSlot = i;
+                            break;
                         }
-                        else
-                        {
-                            PacketSender.SendChatMsg(mPlayer, Strings.Banks.banknospace, ChatMessageType.Bank, CustomColors.Alerts.Error);
-                        }  
                     }
                 }
-                else
+            }
+            else
+            {
+                if (destSlot < 0 || mBank?[destSlot]?.ItemId != Guid.Empty)
                 {
-                    PacketSender.SendChatMsg(mPlayer, Strings.Banks.depositinvalid, ChatMessageType.Bank, CustomColors.Alerts.Error);
+                    destSlot = FindOpenSlot();
+                }
+
+                if (destSlot < 0)
+                {
+                    PacketSender.SendChatMsg(mPlayer, Strings.Banks.banknospace, ChatMessageType.Bank,
+                        CustomColors.Alerts.Error);
+                    return false;
                 }
             }
```

Comment:
I'm wondering if generic and simpler logic is better than introducing an "if 1" condition